src/Domain/CoordinateMaps/Frustum.hpp

```diff
@@ -21,27 +21,246 @@ class er;
+ * x^{+\zeta}_{-\xi,-\eta} + x^{-\zeta}_{+\xi,+\eta} - x^{-\zeta}_{-\xi,-\eta})
+ * \\
+ * \Delta^x_{\xi\zeta} &= \frac{1}{4}(x^{+\zeta}_{+\xi,+\eta} -
+ * x^{+\zeta}_{-\xi,-\eta} - x^{-\zeta}_{+\xi,+\eta} + x^{-\zeta}_{-\xi,-\eta})
+ * \\
+ * \Sigma^y &= \frac{1}{4}(y^{+\zeta}_{-\xi,-\eta} +
+ * y^{+\zeta}_{+\xi,+\eta} + y^{-\zeta}_{-\xi,-\eta} + y^{-\zeta}_{+\xi,+\eta})
+ * \\
+ * \Delta^y_{\zeta} &= \frac{1}{4}(y^{+\zeta}_{-\xi,-\eta} +
+ * y^{+\zeta}_{+\xi,+\eta} - y^{-\zeta}_{-\xi,-\eta} - y^{-\zeta}_{+\xi,+\eta})
+ * \\
+ * \Delta^y_{\eta} &= \frac{1}{4}(y^{+\zeta}_{+\xi,+\eta} -
+ * y^{+\zeta}_{-\xi,-\eta} + y^{-\zeta}_{+\xi,+\eta} - y^{-\zeta}_{-\xi,-\eta})
+ * \\
+ * \Delta^y_{\eta\zeta} &= \frac{1}{4}(y^{+\zeta}_{+\xi,+\eta} -
+ * y^{+\zeta}_{-\xi,-\eta} - y^{-\zeta}_{+\xi,+\eta} + y^{-\zeta}_{-\xi,-\eta})
+ * \\
+ * \Sigma^z &= \frac{z_1 + z_2}{2}\\
+ * \Delta^z &= \frac{z_2 - z_1}{2}
+ * \f}
+ *
+ * The full map is then given by:
+ * \f[\vec{x}(\xi,\eta,\zeta) = \begin{bmatrix}
+ * \Sigma^x + \Delta^x_{\xi}\Xi + (\Delta^x_{\zeta} +
+ * \Delta^x_{\xi\zeta}\Xi)\zeta\\
+ * \Sigma^y + \Delta^y_{\eta}\mathrm{H} + (\Delta^y_{\zeta} +
+ * \Delta^y_{\eta\zeta}\mathrm{H})\zeta\\
+ * \Sigma^z + \Delta^z\zeta\\
+ * \end{bmatrix}\f]
+ *
+ * with Jacobian: \f[J =
+ * \begin{bmatrix}
+ * (\Delta^x_{\xi} + \Delta^x_{\xi\zeta}\zeta)\Xi' &
+ * 0 & \Delta^x_{\zeta}+ \Delta^x_{\xi\zeta}\Xi \\
+ * 0 & (\Delta^y_{\eta} + \Delta^y_{\eta\zeta}\zeta)\mathrm{H}' &
+ * \Delta^y_{\zeta} + \Delta^y_{\eta\zeta}\mathrm{H}\\
+ * 0 & 0 & \Delta^z \\
+ * \end{bmatrix}
+ * \f]
+ *
+ * ### Changing the distribution of gridpoints
+ * By default, the logical coordinate \f$\zeta\f$ is used to linearly
+ * interpolate between the two
+ * bases, resulting in a uniform grid spacing between the bases of the frustum.
+ * This causes elements near the smaller base to the longer in the direction
+ * normal to the base, and elements near the larger base to be shorter in the
+ * direction normal to the base. We desire elements that have roughly equal
+ * sizes along each of their dimensions, and we obtain this through a
+ * \a projective \a map that redistributes the gridpoints in the \f$\zeta\f$
+ * direction.
+ *
+ * \image html BilinearVProjective.png "Comparison of mappings. (Noah Veltman)"
+ *
+ * As seen in Veltman's [Warp-Off]
+ * (https://bl.ocks.org/veltman/8f5a157276b1dc18ce2fba1bc06dfb48),
+ * we can redistribute the gridpoints using a projective map.
+ * We can also see in the figure above that both maps can leave the
+ * elements distorted at one end.
+ * The projective map used in Frustum gets around this difficulty by using
+ * the parameter \f$w_{\delta} = \f$ `projective_scale_factor` to control
+ * the degree of projection, with \f$w_{\delta} = 1\f$ indicating no projection
+ * (bilinear).
+ *
+ * \image html ProjectionOntoPlane.png "Controlling the degree of projection."
+ *
+ * As seen above, \f$w_{\delta}\f$ controls the degree of projection of the
```

Comment:
@markscheel I agree this section has a lot that can be misunderstood and improved upon - I think it would be easier to understand the projection method used to obtain the nonuniform grid if instead I talked about rotating a line into a second spatial dimension, because the concepts and equations obtained are the same. But I don't want to clutter the frustum documentation so much with material that doesn't pertain immediately to the map parameters that a user might be interested in though, so I am considering factoring out a lot of this background geometry stuff into a different documentation page. That way the frustum doc can merely refer to it and future maps using this technique can also simply refer to it without restating everything. I've created an issue for this, #1415 so hopefully I can do a better job of explainin the projection stuff in a page dedicated to just that :smile: Thank you for your helpful comments!